packages/ui/src/forum/modals/CreateThreadModal/machine.ts

```diff
@@ -0,0 +1,61 @@
+import { createMachine, assign } from 'xstate'
+
+import { EmptyObject } from '@/common/types'
+
+export interface GeneralFormContext {
+  topic: string
+  description: string
```

Comment:
For the initial state, the context should have everything optional (unless you'll have data to set initially).

Without that, the `Requred<GeneralFormContext>` will do nothing.

Reply:
Oh, I see below that there's a `Partial<>` as well. I think that we can reducu the number of interfaces to have only two (initial, and later the Transaction one).

Reply:
I wasn't sure about reducing the number of interfaces because I wanted them to have a separate interface with a meaningful name for each step. I'll see if that'll be needed once I've got the transaction step done, and I'll clean them up.

Reply:
OH, I've missed your comment. Sorry for that.

I think clearing them now might be a better option anyway to have clean code and not forget about that in the future ;)